src/metpy/plots/declarative.py

```diff
@@ -699,6 +732,60 @@ def _set_need_redraw(self, _):
         """Watch traits and set the need redraw flag as necessary."""
         self._need_redraw = True
 
+    @staticmethod
+    def _zoom_extent(extent, zoom):
+        """Calculate new bounds for zooming in or out of a given extent.
+
+        ``extent`` is given as a tuple with four numeric values, in the same format as the
+        ``area`` trait.
+
+        If ``zoom`` = 0, the extent will not be changed from what was provided to the method
+        If ``zoom`` > 0, the returned extent will be smaller (zoomed in)
+        If ``zoom`` < 0, the returned extent will be larger (zoomed out)
+
+        """
+        # Measure the current extent
+        center_lon = (extent[0] + extent[1]) / 2
+        center_lat = (extent[2] + extent[3]) / 2
+        lon_range = extent[1] - extent[0]
+        lat_range = extent[3] - extent[2]
+
+        # Transforming zoom by e^-x prevents multiplication by zero or a negative number below
+        zoom_multiplier = np.exp(-0.5 * zoom)
```

Comment:
@kgoebber Do we have any idea what GEMPAK does?

Reply:
I'm looking...don't know exactly where it will be located...hope to update later today.

Reply:
Found it...see below...

For what we are doing it would be for area type (`iartyp`) equal to one, so they have a different zoom multiplier value based on the value of izoom (+/- 1 for each `+` or `-`, respectively).

```fortran
C* Output parameters:                                                   *
C*      IARTYP          INTEGER         Area type                       *
C*                                        -1 = none                     *
C*                                         1 = area name                *
C*                                         2 = center on station        *
C*                                         3 = lat/lon bounds           *
C*                                         4 = DSET                     *
C*                                         5 = @ST                      *
C*                                         6 = @STN1;...;STNN           *
C*                                         7 = @CN:C                    *

        IF  ( iartyp .gt. 0 )  THEN
            IF  ( izoom .le. 0 )  THEN
                zoomul = 2 ** (-izoom)
            ELSE
                zoomul = 1. / (2 ** izoom)
            ENDIF
            IF ( iartyp .eq. 1 ) THEN
                zmofst = (1. - zoomul) / 2.
                rlatll = tlatll + (tlatur - tlatll) * zmofst
                rlatur = tlatur - (tlatur - tlatll) * zmofst
                rlonll = tlonll + (tlonur - tlonll) * zmofst
                rlonur = tlonur - (tlonur - tlonll) * zmofst
            ELSE
                rlatll = cenlat - zoomul * diflat
                rlatur = cenlat + zoomul * diflat
                rlonll = cenlon - zoomul * diflon
                rlonur = cenlon + zoomul * diflon
            ENDIF
        ENDIF
```

Reply:
```fortran
            IF  ( izoom .le. 0 )  THEN
                 zoomul = 2 ** (-izoom)
            ELSE
                zoomul = 1. / (2 ** izoom)
```
🤨 
So they could have just written: `zoomul = 2**-izoom`? Maybe there's some limitation of `**` in FORTRAN I'm not aware of.

Either way, this formulation is entirely the same as the implementation @23ccozad put together (if you do a bit more algebra), with the exception that @23ccozad's uses `exp` instead of `2**`. Powers of two make sense to me here, and overall I like the GEMPAK formulation of:
```fortran
                zmofst = (1. - zoomul) / 2.
                rlatll = tlatll + (tlatur - tlatll) * zmofst
                rlatur = tlatur - (tlatur - tlatll) * zmofst
                rlonll = tlonll + (tlonur - tlonll) * zmofst
                rlonur = tlonur - (tlonur - tlonll) * zmofst
```
better. @23ccozad can you update to match, but using better variable names (like you did before)?